lib/annealing/configuration/configurator.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Annealing
+  class Configuration
+    # Configuration mixin
+    module Configurator
+      def self.included(base)
+        base.send :include, InstanceMethods
+      end
+
+      # Mixin methods
+      module InstanceMethods
+        def initialize(config_hash = {})
```

Comment:
The only problem I see with this implementation is that on each class using the `Configurator` will force the inheritance to this module, I will rather to have a method different name to initialize the config settings by example

```ruby 
  def init_config(config_hash = {})
       @instance_configuration = config_hash
       @temporary_configuration = {}
   end
```
An then one each class extending this behavior will have something like this
```ruby
def initialize(current_state, current_temperature, **config)
  init_config(config)
  ...
end
```

Reply:
OK, sure, that's definitely less surprising that way. I can make that change.